api/v1beta1/disruption_types.go

```diff
@@ -875,6 +883,145 @@ func (s DisruptionSpec) DisruptionCount() int {
+		"just try to delete the disruption. All chaos-injector pods will immediately try to stop the failure.",
+		durationExpl,
+	))
+
+	if s.DryRun {
+		explanation = append(explanation, "spec.dryRun is set to true, meaning we will simulate a real disruption "+
+			"as best as possible, by creating the resource, picking targets, and creating chaos-injector pods, "+
+			"but we will not inject any actual failure.")
+	}
+
+	// s.Level can be "", which defaults to Pod
+	if s.Level != chaostypes.DisruptionLevelNode {
+		explanation = append(explanation, "spec.level is pod. We will pick pods as targets based on your selector, and inject the failure into the pods' containers.")
+	} else {
+		explanation = append(explanation, "spec.level is node. We will pick nodes as targets based on your selector, and inject the failure into the nodes, affecting all pods on those nodes.")
+	}
+
+	if s.Selector != nil {
+		explanation = append(explanation, fmt.Sprintf("This spec has the following selectors which will be used to target %ss with these labels:\n\t%s", s.Level, s.Selector.String()))
+	}
+
+	if s.AdvancedSelector != nil {
+		advancedSelectorExplanation := fmt.Sprintf("This spec has the following advanced selectors which will be used to target %ss based on their labels:\n", s.Level)
+
+		for _, selector := range s.AdvancedSelector {
+			advancedSelectorExplanation += fmt.Sprintf("\t%s\n", selector.String())
+		}
+
+		explanation = append(explanation, advancedSelectorExplanation)
+	}
+
+	if s.Filter != nil && s.Filter.Annotations != nil {
+		explanation = append(explanation, fmt.Sprintf("This spec has the following annotation filters which will be used to target %ss with these annotations.\n\t%s\n", s.Level, s.Filter.Annotations.String()))
+	}
+
+	if s.Containers != nil {
+		explanation = append(explanation, fmt.Sprintf("spec.containers is set, so this disruption will only inject the failure the following containers on the target pods\n\t%s\n", strings.Join(s.Containers, ",")))
+	}
+
+	if s.Pulse != nil {
+		explanation = append(explanation,
+			fmt.Sprintf("spec.pulse is set, so rather than a constant failure injection, after an initial delay of %s"+
+				" the disruption will alternate between an active injected state with a duration of %s,"+
+				" and an inactive dormant state with a duration of %s.\n",
+				s.Pulse.InitialDelay.Duration().String(),
+				s.Pulse.ActiveDuration.Duration().String(),
+				s.Pulse.DormantDuration.Duration().String()))
+	}
+
+	if s.OnInit {
+		explanation = append(explanation, fmt.Sprintf("spec.onInit is true. "+
+			"The disruptions will be launched during the initialization of the targeted pods."+
+			"This requires some extra setup on your end, please [read the full documentation](https://github.com/DataDog/chaos-controller/blob/main/docs/features.md#applying-a-disruption-on-pod-initialization)"))
+	}
+
+	countSuffix := ""
+	if s.Count.Type == intstr.Int {
```

Comment:
here should we also specify that a 100 int value is considered to be 100%? Since it had previously led to confusion for the user

Reply:
It's not! We reject an int value of 100 on create, because users thought it meant 100%. I'll try to clarify that here.

Reply:
yeah I meant the whole behavior around 100 :oops:

Reply:
I can't find the check for this anymore? looking elsewhere...